torch/_jit_internal.py

```diff
@@ -216,6 +229,55 @@ def __getattr__(self, key):
 
     return createResolutionCallbackFromEnv(closure_lookup())
 
+def check_can_compile_named_tuple(cls, loc):
+    """
+    Check that a NamedTuple can be compiled. This means that it must not have additional
+    user-defined methods.
+
+    Arguments:
+        cls: The NamedTuple subclass in question.
+        loc: A SourceRange instance corresponding to the definition of cls.
+    """
+    # These are methods common to all NamedTuples. Using an allowlist is not the cleanest approach,
+    # but these methods don't exist on NamedTuple or tuple, so it is not possible to write code like:
+    #   for name, value in cls.__dict__.items():
+    #       if getattr(NamedTuple, name) is None and getattr(tuple, name) is None:
+    #           # Must be a user-defined function.
+    allowed_methods = {
+        "__annotations__",
+        "__doc__",
+        "__getnewargs__",
+        "__module__",
+        "__new__",
+        "__repr__",
+        "__slots__",
+        "_asdict",
+        "_field_defaults",
+        "_field_types",
+        "_fields",
+        "_make",
+        "_replace",
+        "_source",
+    }
+
+    # Unfortunately, there are some NamedTuples already being used that do have methods that are never
+    # called. Allow these to compile for now.
+    can_compile_allowlist = {
+        torch.nn.utils.rnn.PackedSequence,
```

Comment:
Are there more fb internal unscriptable classes than this one? If so, we can't use allowlist approach.

How about we annotate problematic classes instead of using allowlist? Some ideas:

1) Use @torch.jit.ignore on user-defined methods to make class scriptable
2) Add either a special member like `__torchscript_allow_user_defined_methods` on class

Reply:
I like the ignore approach, let's do that.